pkg/storage/azure/azure.go

```diff
@@ -454,162 +453,228 @@ func (d *driver) StorageChanged(cr *imageregistryv1.Config) bool {
-			return nil
-		}
+		return containerName, true, nil
+	}
 
-		var generatedName bool
-		const numRetries = 5000
-		for i := 0; i < numRetries; i++ {
-			// If the bucket name is blank, let's generate one
-			if len(d.Config.Container) == 0 {
-				// Container name must be between 3 and 63 characters long
-				if d.Config.Container, err = util.GenerateStorageName(d.Listers, ""); err != nil {
-					return err
-				}
-				generatedName = true
-			}
+	if exists, err := d.containerExists(
+		d.Context, environment, d.Config.AccountName, key, d.Config.Container,
+	); err != nil {
+		return "", false, err
+	} else if exists {
+		return d.Config.Container, false, nil
+	}
 
-			err = d.createStorageContainer(environment, d.Config.AccountName, key, d.Config.Container)
-			if err != nil {
-				if e, ok := err.(azblob.StorageError); ok {
-					switch e.ServiceCode() {
-					case azblob.ServiceCodeContainerAlreadyExists:
-						if len(d.Config.Container) != 0 && !generatedName {
-							util.UpdateCondition(cr, defaults.StorageExists, operatorapiv1.ConditionFalse, storageExistsReasonPermissionDenied, "The container exists but we do not have permission to access it")
-							break
-						}
-						d.Config.Container = ""
-						continue
-					default:
-						util.UpdateCondition(cr, defaults.StorageExists, operatorapiv1.ConditionFalse, storageExistsReasonAzureError, fmt.Sprintf("%s", err))
-						return err
-					}
-				} else {
-					util.UpdateCondition(cr, defaults.StorageExists, operatorapiv1.ConditionUnknown, storageExistsReasonAzureError, fmt.Sprintf("Unable to create storage container: %s", err))
-					return err
-				}
-			}
-			cr.Status.StorageManaged = true
-			cr.Status.Storage = imageregistryv1.ImageRegistryConfigStorage{
-				Azure: d.Config.DeepCopy(),
-			}
-			cr.Spec.Storage.Azure = d.Config.DeepCopy()
-			util.UpdateCondition(cr, defaults.StorageExists, operatorapiv1.ConditionTrue, storageExistsReasonContainerExists, "Storage container exists")
+	if err = d.createStorageContainer(
+		environment, d.Config.AccountName, key, d.Config.Container,
+	); err != nil {
+		return "", false, err
+	}
+	return d.Config.Container, true, nil
+}
+
+// verifyUPIConfig verifies if AccountName and Container are present on our configuration.
+func (d *driver) verifyUPIConfig(cr *imageregistryv1.Config) {
+	if d.Config.AccountName == "" {
+		util.UpdateCondition(
+			cr,
+			defaults.StorageExists,
+			operatorapiv1.ConditionFalse,
+			storageExistsReasonNotConfigured,
+			"Storage account key is provided, but account name is not specified",
+		)
+		return
+	}
+
+	if d.Config.Container == "" {
+		util.UpdateCondition(
+			cr,
+			defaults.StorageExists,
+			operatorapiv1.ConditionFalse,
+			storageExistsReasonNotConfigured,
+			"Storage account is provided, but container is not specified",
+		)
+		return
+	}
+
+	cr.Status.StorageManaged = false
```

Comment:
that's not verification, the function name is misleading

Reply:
Yes, makes sense. Updated.